src/backend/distributed/commands/trigger.c

```diff
@@ -0,0 +1,187 @@
+/*-------------------------------------------------------------------------
+ * trigger.c
+ *
+ * This file contains functions to create and process trigger objects on
+ * citus tables.
+ *
+ * Copyright (c) Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+#include "postgres.h"
+#include "distributed/pg_version_constants.h"
+
+#include "access/genam.h"
+#if PG_VERSION_NUM >= PG_VERSION_12
+#include "access/table.h"
+#else
+#include "access/heapam.h"
+#include "access/htup_details.h"
+#endif
+#include "catalog/indexing.h"
+#include "catalog/namespace.h"
+#include "catalog/pg_trigger.h"
+#include "distributed/citus_ruleutils.h"
+#include "distributed/commands.h"
+#include "distributed/listutils.h"
+#include "distributed/metadata_cache.h"
+#include "utils/fmgroids.h"
+#include "utils/lsyscache.h"
+
+/*
+ * GetExplicitTriggerCommands returns the list of DDL commands to (re)create
+ * triggers that are explicitly created for the table with relationId. See
+ * comment of GetExplicitTriggerIds function.
+ */
+List *
+GetExplicitTriggerCommands(Oid relationId)
+{
+	List *createTriggerCommandList = NIL;
+
+	/*
+	 * Set search_path to NIL so that all objects outside of pg_catalog will be
+	 * schema-prefixed. pg_catalog will be added automatically when we call
+	 * PushOverrideSearchPath(), since we set addCatalog to true;
+	 */
+	OverrideSearchPath *overridePath = GetOverrideSearchPath(CurrentMemoryContext);
+	overridePath->schemas = NIL;
+	overridePath->addCatalog = true;
+	PushOverrideSearchPath(overridePath);
+
+	List *triggerIdList = GetExplicitTriggerIds(relationId);
+
+	Oid triggerId = InvalidOid;
+	foreach_oid(triggerId, triggerIdList)
+	{
+		char *createTriggerCommand = pg_get_triggerdef_command(triggerId);
+
+		createTriggerCommandList = lappend(createTriggerCommandList,
+										   createTriggerCommand);
+	}
+
+	/* revert back to original search_path */
+	PopOverrideSearchPath();
+
+	return createTriggerCommandList;
+}
+
+
+/*
+ * GetExplicitTriggerIds returns a list of OIDs corresponding to the triggers
+ * that are explicitly created on the relation with relationId. That means,
+ * this function discards internal triggers implicitly created by postgres for
+ * foreign key constraint validation and the citus_truncate_trigger.
+ */
+List *
+GetExplicitTriggerIds(Oid relationId)
```

Comment:
Minor: `GetExplicitTriggerOidList ` 

I think we typically add `List` to the functions which return a list.